public/app/features/transformers/timeSeriesTable/timeSeriesTableTransformer.ts

```diff
@@ -37,101 +90,179 @@ export const timeSeriesTableTransformer: DataTransformerInfo<TimeSeriesTableTran
  * @remarks
  * For each refId (queryName) convert all time series frames into a single table frame, adding each series
  * as values of a "Trend" frame field. This allows "Trend" to be rendered as area chart type.
- * Any non time series frames are returned as is.
+ *
+ * Any non time series frames are returned unmodified.
  *
  * @param options - Transform options, currently not used
  * @param data - Array of data frames to transform
  * @returns Array of transformed data frames
  *
- * @alpha
+ * @beta
  */
 export function timeSeriesToTableTransform(options: TimeSeriesTableTransformerOptions, data: DataFrame[]): DataFrame[] {
-  // initialize fields from labels for each refId
-  const refId2LabelFields = getLabelFields(data);
-
-  const refId2frameField: Record<string, Field<DataFrameWithValue>> = {};
+  // Initialize maps for labels, sparklines, and reduced values
+  const refId2LabelField: RefFieldMap<string> = {};
+  const refId2FrameField: RefFieldMap<DataFrame> = {};
+  const refId2ValueField: RefFieldMap<number> = {};
 
+  // Accumulator for our final value
+  // which we'll return
   const result: DataFrame[] = [];
 
-  for (const frame of data) {
-    if (!isTimeSeriesFrame(frame)) {
-      result.push(frame);
-      continue;
-    }
+  // Retreive the refIds of all the data
+  let refIdMap = getRefData(data);
 
-    const refId = frame.refId ?? '';
+  // If we're merging data then rather
+  // than creating a series per source
+  // series we initialize fields here
+  // so we end up with one
+  for (const refId of Object.keys(refIdMap)) {
+    const merge = options[refId]?.mergeSeries !== undefined ? options[refId].mergeSeries : MERGE_DEFAULT;
 
-    const labelFields = refId2LabelFields[refId] ?? {};
-    // initialize a new frame for this refId with fields per label and a Trend frame field, if it doesn't exist yet
-    let frameField = refId2frameField[refId];
-    if (!frameField) {
-      frameField = {
-        name: 'Trend' + (refId && Object.keys(refId2LabelFields).length > 1 ? ` #${refId}` : ''),
-        type: FieldType.frame,
-        config: {},
-        values: [],
-      };
-      refId2frameField[refId] = frameField;
+    // Get the frames with the current refId
+    const framesForRef = data.filter((frame) => frame.refId === refId);
```

Comment:
It should show two distinct rows, that's correct! If you don't merge them you should get a series selector for the table with one row in each series.

refId is unique to the query not the frame, as a single query can return multiple frames. This will get all the frames for a single query and proceed from there

Currently, merging is only done per query and not for all frames but it might not be a bad idea to have a "merge all" option 😄

Reply:
right! thanks, got it working now. I was definitely doing something wrong the first time around.